src/Microsoft.AspNetCore.WebUtilities/FormMultipartSection.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    public class FormMultipartSection
+    {
+        private ContentDispositionHeaderValue _contentDispositionHeader;
+
+        public FormMultipartSection(MultipartSection section)
+        {
+            if (!section.IsFormData())
+            {
+                throw new ArgumentException($"Argument must be a form section", nameof(section));
+            }
+
+            Section = section;
+            _contentDispositionHeader = ContentDispositionHeaderValue.Parse(section.ContentDisposition);
```

Comment:
One perf regression here is that we now parse the header multiple times per section, once for IsFile, once for IsFormData and then here. Not sure if there's a simple way to fix that.